### DISCUSSION TITLE
How to handle notebooks which can't be executed on build services

### DISCUSSION THREAD

**Discussion by Cadair** [executablebooks/meta#1007] [Q&A]
Hello!

I have a project I am writing with Jupyterbook and because some of our MyST notebooks need other local software to run, they can't be executed on read the docs (or anywhere else practically). How can we locally build the notebook and publish the outputs on RTD?

If we were using a regular jupyter notebook we could check the outputs in, not ideal but doable. How can we do something similar for MyST notebooks?

**Comment by agoose77** [comment]
We dealt with this over at @scikit-hep/awkward

Whilst you can 'pull' built site from e.g GHA on RTD (with a custom builder), it's not really how it's designed. We found it better to take full ownership of our docs deployment and use an AWS bucket. From my limited research, I think this is increasingly common, with bokeh, and numpy (iirc) as two examples of this pattern.

**Reply by Cadair** [reply]
Thanks for your answer. It's a little sticky as I don't think I will be able to fully build this anywhere but locally, and I would rather not go the extra admin of managing my own site, what RTD gives me does make my life easier, most of the time!
